src/SpeedDial/SpeedDial.js

```diff
@@ -0,0 +1,144 @@
+// @flow weak
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { withStyles } from 'material-ui/styles';
+import Button from 'material-ui/Button';
+import Icon from 'material-ui/Icon';
+import AddIcon from 'material-ui-icons/Add';
+
+const styles = theme => ({
+  root: {
+    width: 56,
+    textAlign: 'center',
+    zIndex: theme.zIndex.speedDial,
+    display: 'flex',
+    flexDirection: 'column-reverse', // Place the Actions above the FAB.
+  },
+  fab: {
+    transition: theme.transitions.create('transform', {
+      duration: theme.transitions.duration.shortest,
+    }),
+  },
+  hiddenFab: {
+    transform: 'scale3d(0, 0, 1)',
+  },
+  fabIcon: {
+    transition: theme.transitions.create('transform', {
+      duration: theme.transitions.duration.shorter,
+      easing: theme.transitions.easing.easeInOut,
+    }),
+  },
+  openFabIcon: {
+    transform: 'rotate(135deg)',
+  },
+  actions: {
+    display: 'flex',
+    flexDirection: 'column-reverse', // Display the first action at the bottom.
+    marginBottom: 16,
+  },
+  closedActions: {
+    transition: 'top 0s linear 0.2s',
+  },
+});
+
+function SpeedDial(props) {
+  const {
+    children: childrenProp,
+    classes,
+    className: classNameProp,
+    hidden,
+    icon: iconProp,
+    label,
+    open,
+    onClick,
+    ...other
+  } = props;
+
+  const rootClassName = classNames(classes.root, classNameProp);
+  const buttonClassName = classNames(classes.fab, hidden && classes.hiddenFab);
+  const fabIconClassName = classNames(classes.fabIcon, open && classes.openFabIcon);
+  const actionsClassName = classNames(classes.actions, !open && classes.closedActions);
+
+  const children = React.Children.map(childrenProp, (child, childIndex) => {
+    if (!React.isValidElement(child)) return null;
+    const delay = 30 * (open ? childIndex : childrenProp.length - childIndex);
+    return React.cloneElement(child, { delay, open });
+  });
+
+  let icon = null;
+  if (iconProp) {
+    if (React.isValidElement(iconProp)) {
+      icon = iconProp;
+    } else if (typeof iconProp === 'string') {
+      icon = <Icon>{iconProp}</Icon>;
+    }
+  } else {
+    icon = <AddIcon className={fabIconClassName} />;
+  }
+
+  return (
+    <div className={rootClassName} {...other}>
+      <Button
+        id={label}
```

Comment:
Humm. I'm wondering about id escaping here. `string -> id` might not be as straightforward.

Reply:
Mmm... good point. I originally had a separate prop, but was trying to keep the API surface down. I've added a regex to filter the label.